lucene/queries/src/java/org/apache/lucene/queries/function/valuesource/QueryValueSource.java

```diff
@@ -139,15 +138,14 @@ public boolean exists(int doc) {
     lastDocRequested = doc;
 
     try {
-      if (noMatches) return false;
       if (scorer == null) {
```

Comment:
This now needs to check if disi isn't null, otherwise we might call weight.scorer a ton!

Reply:
🤦🏻‍♂️